frontend/src/components/graph/Constants.ts

```diff
@@ -22,3 +24,11 @@ export interface SubDagFlowElementData extends FlowElementDataBase {
   // Callback action if a SubDag expand button is clicked.
   expand: (nodeKey: string) => void;
 }
+
+export interface ExecutionFlowElementData extends FlowElementDataBase {
```

Comment:
nit: This is purely a data type.
Usually I use typescript type syntax for data types and only use interface for sth that can be a class.
You can do the following:
```
export type A = B & C
```
It means A type has all the fields from B type and C type.

Reply:
I have been thinking about interface vs type for some time myself as well. Based on https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces, I think both are very similar, and `type` is more strict from the `extension` capability. Sounds good on using `type` here.